test/mail_helper.js

```diff
@@ -104,46 +104,53 @@ module.exports = (printLogs) => {
     // HTTP half
 
     var hapi = require('hapi')
-    var api = new hapi.Server()
-    api.connection({
-      host: config.smtp.api.host,
-      port: config.smtp.api.port
+    var api = new hapi.Server({
+        host: config.smtp.api.host,
+        port: config.smtp.api.port
     })
 
-    function loop(email, cb) {
-      var mail = users[email]
-      if (! mail) {
-        return setTimeout(loop.bind(null, email, cb), 50)
-      }
-      cb(mail)
+    function loop(email) {
+      return new Promise((res) => {
+        var mail = users[email]
+        while ((! mail)) {
+          new Promise((res) => {
+            setTimeout(res, 50)
+          })
+          mail = users[email]
+        }
+        res(mail)
+      }, reject)
     }
 
     api.route(
       [
         {
           method: 'GET',
           path: '/mail/{email}',
-          handler: function (request, reply) {
-            loop(
-              decodeURIComponent(request.params.email),
-              function (emailData) {
-                reply(emailData)
-              }
-            )
+          handler: async function (request, h) {
+             return loop(
+               decodeURIComponent(request.params.email)
+              )
+             .then(() => {
+               return emailData
+             })
+             .catch((err) => {
+               return err
+              })
           }
         },
         {
           method: 'DELETE',
           path: '/mail/{email}',
-          handler: function (request, reply) {
+          handler:async function (request, h) {
             delete users[decodeURIComponent(request.params.email)]
-            reply()
+            return;
```

Comment:
don't forget to apply the diff I sent you here, seems like this needs a `return {}`